docs: editing pass on getting-started (#6703)
apollographql/apollo-server: docs/source/getting-started.mdx

## Step 2: Install dependencies
Applications that run Apollo Server require two top-level dependencies:
* [`graphql`](https://npm.im/graphql) (also known as `graphql-js`) is the library that implements the core GraphQL parsing and execution algorithms.

> * [`@apollo/server`](https://www.npmjs.com/package/@apollo/server) is the main library for Apollo Server itself, which understands how to turn HTTP requests and responses into GraphQL operations, and runs them in an extensible context with support for plugins and other features.

Reviewer: I feel like we should break this down into two sentences, like so:

```suggestion
* [`@apollo/server`](https://www.npmjs.com/package/@apollo/server) is the main library for Apollo Server itself. Apollo Server knows how to turn HTTP requests and responses into GraphQL operations and run them in an extensible context with support for plugins and other features.
```

For more information on the compiler options above, see the [TypeScript Compiler docs](https://www.typescriptlang.org/tsconfig).

> 5. Finally, replace the default `scripts` entry in your `package.json` file with these `type` and `scripts` entries:

Reviewer: ```suggestion
5. Finally, replace the default `scripts` entry in your `package.json` file with the following `type` and `scripts` entries:
```

The above `start` script tells TypeScript to compile your code into JavaScript before using `node` to run that compiled code. Setting your project's [`type` to `module`](https://nodejs.org/api/packages.html#approach-1-use-an-es-module-wrapper) loads your JavaScript files as ES modules, enabling you to use top-level [`await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) calls.

> 6. You can now run `npm start`, which should successfully compile and run your empty `index.ts` file, printing something like the following:

Reviewer: ```suggestion
6. You can now run `npm start`, which should successfully compile and run your empty `index.ts` file, printing something like this:
```

  touch index.js
```

Now replace the default `scripts` entry in your `package.json` file with these `type` and `scripts` entries:
```json title="package.json"
{
  // ...etc.
  "type": "module",
  "scripts": {

>     "start": "tsc && node ./dist/index.js"

Reviewer: We shouldn't need to compile with TS here, so this should be:
```suggestion
    "start": "node index.js"
```

Author: Oops, good catch! (We do need the `type` line though, or we could name the file index.mjs.)

## Combined example

You can view and fork this complete example on CodeSandbox:
<!-- TODO(AS4): Fix this sandbox link after docs examples is up, and make sure it matches the final version -->
<a href="https://codesandbox.io/s/github/apollographql/docs-examples/tree/RK/add-as4-getting-started/apollo-server/v4/getting-started?fontsize=14&hidenavigation=1&theme=dark">
  <img alt="Edit server-getting-started" src="https://codesandbox.io/static/img/play-codesandbox.svg" />
</a>

## Next steps

> TODO(AS4): Should we temporarily remove this while we don't have full docs?

Reviewer: This is a _really good_ question that I went back and forth on. 

On one hand, I do feel like providing these resources is helpful for those looking for more information about the basics of GraphQL servers (particularly the "Schema basics" and "Resolvers" articles).  On the other hand, we could wait and reintroduce these when we know they align with the AS4 syntax, so folks don't get confused. I could go either way, @glasser which would you prefer?

Author: I think we should make a change to the top level index.mdx page that's like "as4 is in alpha, we've removed most of the docs that haven't been updated yet, you can trust anything actually here is up to date and otherwise go look at AS3, most stuff that isn't addressed explicitly in the migration guide probably still works"

then idk what to do exactly here: I'd kinda like to avoid broken links though?

Reviewer: Oh, the below links won't be broken on the main site -> they'll just lead you to the "main" Apollo Server docset, which would be AS3. So `/apollo-server/schema/schema/` should navigate to `https://www.apollographql.com/docs/apollo-server/schema/schema/`

Reviewer: I'll make a PR for adding a note to the top-level index.mdx page now
